python/katana_setup.py

```diff
@@ -143,11 +143,44 @@ def load_lang_config(lang):
         return dict(compiler=[], linker=[], extra_compile_args=[], extra_link_args=[], include_dirs=[])
     parser = configparser.ConfigParser()
     parser.read(filename, encoding="UTF-8")
+    linker_wrapper_flag = split_cmake_list(parser.get("cmake", "LINKER_WRAPPER_FLAG", fallback="-Wl,"))
+    linker_wrapper_flag_sep = parser.get("cmake", "LINKER_WRAPPER_FLAG_SEP", fallback=",")
+
```

Comment:
probably a short description of the *ahem* workaround and maybe an example here to help future readers orient themselves, e.g.,
"""
...KatanaPythonSetupSubdirectory.cmake generates a file that contains build and link flags that we read back here.... They typically look like: ....

TODO(amp): Starting with CMake 3.20, these flags contain `LINKER:` and `SHELL:` generator expressions which we don't know how to parse generally.
"""

Reply:
BTW, there was an update on the discourse issue you raised: https://discourse.cmake.org/t/generating-linker-command-line-arguments-to-a-file/3117/2 though I don't think the the answer is super satisfying.

Reply:
Yes. I saw it. It just made me hate CMake. The generator layer is so opaque and almost impossible to get useful information out of except by running the build.